core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -693,6 +754,25 @@ public Custom getCustom(String type) {
             return this.customs.get(type);
         }
 
+        public Builder putActiveAllocationIds(int shardId, Set<String> allocationIds) {
+            activeAllocationIds.put(shardId, Collections.unmodifiableSet(new HashSet(allocationIds)));
```

Comment:
wondering if we should enforce immutability on this level... feels more natural to do it in the build()